kong/include/kong/pluginsocket.proto

```diff
@@ -328,7 +328,7 @@ service Kong {
     rpc Router_GetRoute(google.protobuf.Empty) returns (Route);
     rpc Router_GetService(google.protobuf.Empty) returns (Service);
 
-    rpc Service_SetUpstream(String) returns (google.protobuf.Empty);
```

Comment:
Does it work with only `Bool` ? The general PDK says that there are 2 values returned: (bool, string) - https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.service/#kongserviceset_upstreamhost

Reply:
Good question!
I have verified the change works for both cases.

Longer answer: 
Although it fixes the behavior described in https://github.com/Kong/go-pdk/issues/114, the go-pdk does not currently handle consistently the `nil, err` protocol adopted in the Lua PDK -- ie, most methods do not return the `err` on the Go PDK side (only other errors are reported, like connection errors, protobuf errors, etc). Also, on the Go PDK side the `SetUpstream` method does not return a boolean, only an `error`. Fixing this means an API breaking change has to be introduced, so this really needs to be done before we release `v1.0.0`, but not in the scope of this fix. I will take a look of this.

Reply:
Oh, ok! That makes sense. Thanks 👍 !